test/e2e/performanceprofile/functests/1_performance/cpu_management.go

```diff
@@ -232,58 +231,34 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", Ordered, func() {
 	When("pod runs with the CPU load balancing runtime class", func() {
 		var smtLevel int
 		var testpod *corev1.Pod
-		var defaultFlags map[int][]int
 
-		getCPUsSchedulingDomainFlags := func() (map[int][]int, error) {
-			cmd := []string{"/bin/bash", "-c", "more /proc/sys/kernel/sched_domain/cpu*/domain*/flags | cat"}
-			out, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
+		getCPUsSchedulingDomains := func() ([]string, error) {
+			cmd := []string{"/bin/bash", "-c", "cat /proc/schedstat"}
+			schedstat, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
 			if err != nil {
 				return nil, err
 			}
-
-			re, err := regexp.Compile(`/proc/sys/kernel/sched_domain/cpu(\d+)/domain\d+/flags\n:+\n(\d+)`)
-			if err != nil {
-				return nil, err
-			}
-
-			allSubmatch := re.FindAllStringSubmatch(out, -1)
-			cpuToSchedDomains := map[int][]int{}
-			for _, submatch := range allSubmatch {
-				if len(submatch) != 3 {
-					return nil, fmt.Errorf("the sched_domain submatch %v does not have a valid length", submatch)
-				}
-
-				cpu, err := strconv.Atoi(submatch[1])
-				if err != nil {
-					return nil, err
-				}
-
-				if _, ok := cpuToSchedDomains[cpu]; !ok {
-					cpuToSchedDomains[cpu] = []int{}
-				}
-
-				flags, err := strconv.Atoi(submatch[2])
-				if err != nil {
-					return nil, err
+			lines := strings.Split(schedstat, "\n")
+			cpusWithoutDomainLines := []string{}
+
+			for i := 0; i < len(lines); i++ {
+				line := lines[i]
+				if strings.HasPrefix(line, "cpu") {
+					if i+1 >= len(lines) || !strings.HasPrefix(lines[i+1], "domain") {
```

Comment:
The if is hard to read written like this (two negative checks with OR). I understand what it is doing, but I think it needs a comment.

Reply:
Added the comments in the latest commit.